spring-boot-tools/spring-boot-loader-tools/src/main/resources/org/springframework/boot/loader/tools/launch.script

```diff
@@ -116,9 +116,17 @@ PID_FOLDER="$PID_FOLDER/${identity}"
 pid_file="$PID_FOLDER/{{pidFilename:${identity}.pid}}"
 log_file="$LOG_FOLDER/$LOG_FILENAME"
 
-# Determine the user to run as if we are root
-# shellcheck disable=SC2012
-[[ $(id -u) == "0" ]] && run_user=$(ls -ld "$jarfile" | awk '{print $3}')
+# The application will use the user specified via
+if [[ -z "{{runUser}}" ]]; then
+    # Determine the user to run as if we are root
+    # shellcheck disable=SC2012
+    [[ $(id -u) == "0" ]] && run_user=$(ls -ld "$jarfile" | awk '{print $3}')
+else
+    run_user="{{runUser}}"
+fi
+
+# Issue an warning if the application will run as root
+[[ $(id -u) == "0" ]] && { echoYellow "[WARN] Application is running as root (UID 0). This is considered insecure."; }
```

Comment:
Shouldn't this be `$(id -u $run_user)`?

Reply:
Well spotted, @kartoffelsup. I think you're right. Thank you.

That raises the issue of adding some tests for this which I overlook when I said that I didn't think that there was anything left to be done.

@obfischer do you have the time to look at adding some tests please? I'm happy to guide you through the process as needed.

Reply:
Yes, of course. Which tests are missing from your perspective?

Reply:
A couple of tests would be good:

1. Verify that the warning is generated when the application will run as root
2. Verify that the warning is not generated when the application will run as another user

Reply:
I think the correct module for such tests will be `spring-boot-integration-tests`?

Reply:
Yes, that's right. Thanks.

Reply:
I will start now with it.